doc/harmony-ap_logging_user_guide.md

```diff
@@ -0,0 +1,371 @@
+
+```Dockerfile
+FROM niis/harmony-ap:2.3.0
+
+COPY logback.xml /opt/harmony-ap/setup/logback.xml
+```
+
+When the container starts, the file is moved from the `setup` directory to `/var/opt/harmony-ap/etc/logback.xml`.
+
+## 4 Enable logging of full messages for debugging
+
+By default, AP logs do not include messages sent/received by the Access Point. In some situations, logging the full messages might be required for debugging purposes. In that case, the logging configuration can be changed.
+
+The following steps describe how to enable logging of full messages in Harmony AP:
+
+1. Open the Logback configuration file (`logback.xml`). Refer to the section on modifying the Logback configuration for the [Linux package](#312-modifying-the-logback-configuration) or [the Docker image](#322-modifying-the-logback-configuration) to locate the file.
+2. Locate the logger configuration for the `org.apache.cxf` package.
+3. Change the log level from `WARN` to `INFO`:
+   ```xml
+   <!-- In order to enable logging of request/responses please change the loglevel to INFO -->
+   <logger name="org.apache.cxf" level="INFO"/>
+   ```
+
+## 5. Centralized logging
+
+To monitor the system's status, users typically need to manually access the logs on the machine running AP. While this approach may suffice for some systems, it can be inefficient for others, particularly when managing multiple AP instances.
+
+A centralized logging system is recommended to address this potential issue. It aggregates logs from one or more services, including AP instances, into a single platform. This approach simplifies system performance monitoring, troubleshooting, and log analysis.
+
+The ELK stack is a popular solution for centralized logging, and this documentation focuses on its implementation. ELK combines tools for collecting, parsing, and visualizing logs from multiple sources. It consists of three components:
+
+- **Elasticsearch**: A distributed search and analytics engine for storing and indexing log data.
+- **Logstash**: A data processing pipeline for ingesting, processing, and forwarding logs to Elasticsearch.
+- **Kibana**: A visualization tool for exploring and analyzing log data stored in Elasticsearch.
+
+The following diagram illustrates the flow of logs from AP to the ELK stack:
+
+![AP with ELK](img/ug_ap_logging_elk.svg)
+
+### 5.1 Sending logs from AP to Elasticsearch with Logstash
+
+To set up the ELK stack, the first step is installing and configuring Logstash. This section details how to configure Logstash to receive logs from AP and send them to Elasticsearch for storage and analysis. If the ELK stack isn't installed, refer to the provided [Docker setup example](#54-example-docker-environment-of-harmony-ap-with-elk-stack) for a quick start.
```

Comment:
> If the ELK stack isn't installed, refer to the provided [Docker setup example](#54-example-docker-environment-of-harmony-ap-with-elk-stack) for a quick start.

For clarity, it would be good to specify that the provided example setup is meant for testing purposes only and the official documentation provided by Elastic should be used when setting up a production level setup.